paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/lookup/PrimaryKeyPartialLookupTable.java

```diff
@@ -108,25 +109,31 @@ public void open() throws Exception {
 
     @Override
     public List<InternalRow> get(InternalRow key) throws IOException {
+        Triple<BinaryRow, Integer, InternalRow> partitionAndBucket = extractPartitionAndBucket(key);
+        InternalRow kv =
+                queryExecutor.lookup(
+                        partitionAndBucket.f0, partitionAndBucket.f1, partitionAndBucket.f2);
+        if (kv == null) {
+            return Collections.emptyList();
+        } else {
+            return Collections.singletonList(kv);
+        }
+    }
+
+    private synchronized Triple<BinaryRow, Integer, InternalRow> extractPartitionAndBucket(
+            InternalRow key) {
         InternalRow adjustedKey = key;
         if (keyRearrange != null) {
             adjustedKey = keyRearrange.replaceRow(adjustedKey);
```

Comment:
It is reused

Reply:
I have encapsulated these as a separate function called `extractPartartitionAndBucket`. And it has been declared as synchronized, so there will be no thread safety issues.

Reply:
I have encapsulated these as a separate function called extractPartartitionAndBucket. And it has been declared as synchronized, so there will be no thread safety issues.